src/actions/motion.ts

```diff
@@ -53,15 +53,12 @@ abstract class MoveByScreenLine extends BaseMovement {
     vimState: VimState,
     count: number
   ): Promise<Position | IMovement> {
-    if (
-      vimState.currentMode === Mode.Visual &&
-      vimState.cursorStopPosition.isAfterOrEqual(vimState.cursorStartPosition)
-    ) {
+    if (vimState.currentMode === Mode.Visual) {
       // The selection is on the right side of the cursor, while our representation considers the cursor to be the left edge
-      vimState.editor.selection = new vscode.Selection(
-        vimState.cursorStartPosition,
-        vimState.cursorStopPosition
-      );
+      const multicursorIndex = this.multicursorIndex ?? 0;
+      let selections = vimState.editor.selections;
```

Comment:
Any reason `vimState.editor.selections[multicursorIndex] = ...` doesn't work?

Reply:
I remember that when I did this I've made like you propose and it didn't work. But now that I'm looking more into this I believe this all "MoveByScreenLine" has quite a few bug and needs some change. I'm working on this.

Reply:
This 'cursorMove' command moves the cursor on all multicursors at the same time, so we can only run once on the main cursor (index 0). Then on the other cursors we should only get the updated position. However sometimes the selections combine and then we'll have more cursors than vscode has selections, so we can't update a position of a cursor so I'm removing those cursors.
Example: If we have two cursors with two selections, if we move them in a way that they combine, vscode will have only one selection. But our 'MoveByScreenLine' will still be called for 'multicursorIndex' 1, when we see that it doesn't exist anymore, we return 'IMovement' with a new info `removed: true` so that we can at the end of 'executeMovement' remove the cursors that no longer exist.

If tests are ok I'll push this fix soon.